File: GoGolang/one.go
package main
import "fmt"
func main(){
	var S, N, M int
	fmt.Print("Введите значение S: ")
	fmt.Scan(&S)
	fmt.Print("Введите значение N: ")
	fmt.Scan(&N)
	fmt.Print("Введите значение M: ")
	fmt.Scan(&M)
	result := days(S, N, M)
	if result == -1 {
		println("Выжить невозможно")
	} else {
		println("Минимальное количество дней: ", result)
	}
}
func days(S, N, M int) int{
	if M > N {
		return -1
	}
	sundays := S / 7
	need_food := S * M
	shop_work := S - sundays
	minimal := (need_food + N - 1) / N
	if minimal > shop_work{
		return -1 
	}
	return minimal
}

File: ++/one.cpp
#include <iostream>
using namespace std;
int ostrov(int S, int M, int N)
{
    if (M > N)
    return -1;
    int sundays = S / 7;
    int shop_work = S - sundays;
    int need_food = S * M;
    int min_days = (need_food + N - 1) / N;
    if (min_days > shop_work)
    return -1;
return min_days;
}
int main()
{
    int S, N, M;
    cout << "Введите число S: ";
    cin >> S;
    cout << "Введите число N: ";
    cin >> N;
    cout << "Введите число M: ";
    cin >> M;
    int result = ostrov(S, M, N);
    if (result == -1 )
    cout << "Выжить невозможно" << endl;
    else
    cout << "Минимальное количество дней: " << result << endl;
}
